tenprintcover.py

```diff
@@ -498,25 +498,50 @@ def drawShape(c, x, y, s):
         else:
             assert not "Implement."
 
-
+    # If the text is long, use a smaller font size. 
+    def scale_font(text, font_name, font_properties):
+        (font_size, font_slant, font_weight) = font_properties
+        width = len(text) * font_size
+        if width > cover_width * 7:   # This is an empirical, unintelligent, heuristic.
+            return  (font_size * 0.8, font_slant, font_weight)
+        else:
+            return font_properties
+    
+    # Return a font appropriate for the text. Uses Noto CJK if text contains letters of
+    # Simplified Chinese, Traditional Chinese, Japanese, and Korean (CJK), otherwise Noto Sans. 
+    # http://www.unicode.org/faq/han_cjk.html
+    def select_font(text):
+        for char in text:
+            if ord(char) >= 0x4E00:
+                return 'Noto Sans CJK SC'
+        return 'Noto Sans'
```

Comment:
If you would like to check this code in as-is, then yes, we should open a high-pri issue and address it soon. Personally, I would prefer using a regex with Unicode Script support and avoid hard-wired constants.

Note though that by looking for Unicode point `4E00` and above, you miss other ranges below (e.g. [CJK Unified Ideographs Extension A, from `3400`..`4DBF`](https://unicode-table.com/en/blocks/cjk-unified-ideographs-extension-a/), and include non-CJK scripts that start at `A000` (the [Yi Syllables](https://unicode-table.com/en/blocks/yi-syllables/)).